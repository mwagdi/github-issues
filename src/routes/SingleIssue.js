import React,{ Component } from 'react';
import { connect } from 'react-redux';
import axios from 'axios';

import Comment from '../components/Comment';

import { API } from '../constants';
import { getComments } from '../store/actions';

class SingleIssue extends Component{
    constructor(props){
        super(props);
        this.state = {
            issue: {}
        }
    }
    componentDidMount(){
        axios.get(`${API}/${this.props.match.params.number}`)
        .then(response => {
            this.setState({ issue: response.data });
            this.props.getComments(response.data.comments_url);
        });
    }
    render(){
        const { issue } = this.state;
        const { comments } = this.props;
        return(
            <div className="singleIssue">
                <div className="row">
                    <div className="singleIssue__header">
                        <h1 className="singleIssue__title">{issue.title}</h1>
                        <h2 className="singleIssue__number">#{issue.number}</h2>
                    </div>
                    <div className="flex-container">
                        <div className="comments-container flex-grow">
                            {comments.map((comment,i) => (
                                <Comment key={i} comment={comment} />
                            ))}
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => {
    return {
        comments: state.comments
    }
}

export default connect(mapStateToProps,{ getComments })(SingleIssue);