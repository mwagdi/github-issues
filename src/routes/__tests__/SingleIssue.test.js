import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import SingleIssue from '../SingleIssue';

test('SingleIssue renders correctly', () => {
    const match = {
        params: {
            number: 3
        }
    }
    const store = {
        getState: () => ({
            comments: [],
        }),
        dispatch: () => { },
        subscribe: () => { },
    };
    const wrapper = shallow(<SingleIssue match={match} store={store} />).dive();
    expect(toJSON(wrapper)).toMatchSnapshot();
})
