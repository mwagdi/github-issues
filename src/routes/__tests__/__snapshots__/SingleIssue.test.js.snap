// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleIssue renders correctly 1`] = `
<div
  className="singleIssue"
>
  <div
    className="row"
  >
    <div
      className="singleIssue__header"
    >
      <h1
        className="singleIssue__title"
      />
      <h2
        className="singleIssue__number"
      >
        #
      </h2>
    </div>
    <div
      className="flex-container"
    >
      <div
        className="comments-container flex-grow"
      />
    </div>
  </div>
</div>
`;
